src/chaat/model/user.clj

```diff
@@ -0,0 +1,72 @@
+(ns chaat.model.user
+  (:require
+   [chaat.db.user :as db.user]
+   [crypto.password.bcrypt :as bcrypt]
+   [java-time.api :as jt]
+   [next.jdbc.date-time :as dt]
+   [chaat.errors :refer [do-or-error]]))
+
+;; Model layer does validation pertaining to the representation 
+;; of a user: username & password format (length, special characters etc.)
+
+(defn min-length?
+  "Check if string is >= a threshold length"
+  [str threshold]
+  (>= (count str) threshold))
+
+;; Add more restrictions: special characters not allowed, only alphanumeric etc.
+;; Perhaps I can start using the schema library? Or implement something similar myself.
+(defn validate-username-format
+  "Basic format check for username"
+  [username]
+  (let [min-length 2]
+    (if (min-length? username min-length)
+      {:result username :error nil}
+      {:result nil :error "Wrong username format"})))
+
+(defn validate-password-format
+  "Basic format check for password"
+  [password]
+  (let [min-length 8]
+    (if (min-length? password min-length)
+      {:result password :error nil}
+      {:result nil :error "Wrong password format"})))
+
+(defn validate-signup-details
+  "Basic format check for signup details: username & password"
+  [username password]
+  (let [result (validate-username-format username)
+        result (do-or-error result validate-password-format password)]
+    result))
+
+(defn get-time-instant
```

Comment:
question: this fn also seems redundant, any reason for doing this?

Reply:
We forgot to talk about this during our call.
I needed to stub the time function for some of my tests, and it was easier to stub get-time-instant rather than jt/instant. That way I could still use the original functionality of jt/instant during my tests.